core/src/autogluon/core/constants.py

```diff
@@ -14,5 +14,6 @@
 # AG_ARGS variables are key names in model hyperparameters to dictionaries of custom AutoGluon arguments.
 AG_ARGS = 'AG_args'  # Contains arguments to control model name, model priority, and the valid configurations which it can be used in.
 AG_ARGS_FIT = 'AG_args_fit'  # Contains arguments that impact model training, such as early stopping rounds, #cores, #gpus, max time limit, max memory usage  # TODO
+AG_ARGS_STACK = 'AG_args_stack'  # Contains arguments that impact model stacking, such as if the model is allowed to be a stacker, or if it is allowed to use the original features  # TODO: v0.1 add to documentation
```

Comment:
How advantageous is it to have 3 different dicts here vs just one `AG_ARGS` dict that holds all of these options?  To me the `hyperparameters` argument now seems have an extremely deep amount of dictionary/list nesting.

Reply:
I agree. Is it possible to merge some options together? May be in a later PR since it will also impact `AG_ARGS_FIT`.

Reply:
AG_ARGS_FIT should definitely remain its own separate thing since it is information which is visible to the model. The main question is if AG_ARGS_STACK should be combined with AG_ARGS.

Looking deeper into it, I think that is a fair request.

ex:

In AG_ARGS there exist these options:
-  problem_types: (list) List of valid problem types for the model. `problem_types=['binary']` will result in the model only being trained if `problem_type` is 'binary'.
- disable_in_hpo: (bool) If True, the model will only be trained if `hyperparameter_tune=False`.

These are very similar to the 'valid_stacker' option, so I think it makes sense.

However, there is one aspect which isn't covered here that I haven't implemented and that is specification of the stack/bagged ensemble hyperparameters (not the inner model hyperparameters), which ideally could be different for every model. This probably lives best in some kind of AG_ARGS_STACK key. Thoughts?

Reply:
I see your point re additional ensembling hyperparameters, so I guess having the extra dict is ok to maintain separation of concerns. But I'd consider renaming this dict to `AG_ARGS_ENSEMBLE` so it at least encompasses all ensembling hyperparams, and we don't need separate dicts for bagging hyperparams, stacking hyperparams, etc.

I think the `fit()` docs should only have one sentence describing these `AG_ARGS*` dicts, with a link to another static page on our website that describes these dicts in detail. Otherwise it's information overload.

Reply:
Added AG_ARGS_ENSEMBLE. Regarding the docs, do you have a suggestion for how we ensure they stay aligned such that an old version doesn't link to a new versions docs?